File: domain/src/main/java/com/omvp/domain/interactor/impl/GetLocaleUseCaseImpl.java
package com.omvp.domain.interactor.impl;

import com.omvp.domain.interactor.GetLocaleUseCase;
import com.omvp.domain.repository.LocaleRepository;

import java.util.Locale;

import javax.inject.Inject;

import io.reactivex.Single;

public class GetLocaleUseCaseImpl extends BaseUseCaseImpl<LocaleRepository> implements GetLocaleUseCase {

    @Inject
    GetLocaleUseCaseImpl(LocaleRepository repository) {
        super(repository);
    }

    @Override
    public Single<Locale> execute() {
        return getRepository().retrieve();
    }

}

File: domain/src/main/java/com/omvp/domain/interactor/impl/GetLocaleUseCaseImpl.kt
package com.omvp.domain.interactor.impl

import com.omvp.domain.interactor.GetLocaleUseCase
import com.omvp.domain.repository.LocaleRepository

import java.util.Locale

import javax.inject.Inject

import io.reactivex.Single

class GetLocaleUseCaseImpl @Inject
internal constructor(repository: LocaleRepository) : BaseUseCaseImpl<LocaleRepository>(repository), GetLocaleUseCase {

    override fun execute(): Single<Locale> {
        return repository.retrieve()
    }

}
